src/components/BareMetalHosts/BaremetalHostStatus.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getHostStatus } from '../../selectors/host/selectors';
+import { GenericStatus, GenericError, GenericProgress, GenericSuccess, ValidationError } from './StatusComponents';
+
+// Map status strings to components
+const componentMap = {
+  'validation error': ValidationError,
+  'registration error': GenericError,
+  registering: GenericProgress,
+  inspecting: GenericProgress,
+  discovered: GenericSuccess,
+  OK: GenericSuccess,
+  ready: GenericSuccess,
+  provisioned: GenericSuccess,
+  provisioning: GenericProgress,
+  'preparing to provision': GenericProgress,
+  deprovisioning: GenericProgress,
+  'making host available': GenericProgress,
+};
+
+export const BaremetalHostStatus = ({ host }) => {
+  const hostStatus = getHostStatus(host);
```

Comment:
we found out that for better reusability it is better to decouple status functionality from the UI (e.g. vmStatus is used at many places). Please see  `utils/status`. Could you use this approach here?

 Basically it is a function which returns

```js
{
   status, // constant
   ...additionalUsefulData // e.g. message
}
```